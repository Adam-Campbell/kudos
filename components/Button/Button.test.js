import { shallow } from 'enzyme';
import { Button, AnchorButton } from './index';

describe('<Button/>', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <Button>A Button</Button>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('the isHovered prop is toggled appropriately on MouseEnter and MouseLeave',() => {
        const wrapper = shallow(
            <Button>A Button</Button>
        );
        expect(wrapper.find('ButtonElement').prop('isHovered')).toBe(false);
        wrapper.find('ButtonElement').simulate('mouseEnter');
        expect(wrapper.find('ButtonElement').prop('isHovered')).toBe(true);
        wrapper.find('ButtonElement').simulate('mouseLeave');
        expect(wrapper.find('ButtonElement').prop('isHovered')).toBe(false);  
    });
    test('any additional props are passed to the rendered output correctly', () => {
        const wrapper = shallow(
            <Button foo="bar">A Button</Button>
        );
        expect(wrapper.find('ButtonElement').prop('foo')).toBe('bar');
    });
});

describe('<AnchorButton/>', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <AnchorButton>An Anchor Button</AnchorButton>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test("the isHovered prop is toggled appropriately on MouseEnter and MouseLeave",() => {
        const wrapper = shallow(
            <AnchorButton>An Anchor Button</AnchorButton>
        );
        expect(wrapper.find('ButtonElement').prop('isHovered')).toBe(false);
        wrapper.find('ButtonElement').simulate('mouseEnter');
        expect(wrapper.find('ButtonElement').prop('isHovered')).toBe(true);
        wrapper.find('ButtonElement').simulate('mouseLeave');
        expect(wrapper.find('ButtonElement').prop('isHovered')).toBe(false);  
    });
    test('any additional props are passed to the rendered output correctly', () => {
        const wrapper = shallow(
            <AnchorButton foo="bar">An Anchor Button</AnchorButton>
        );
        expect(wrapper.find('ButtonElement').prop('foo')).toBe('bar');
    });
});