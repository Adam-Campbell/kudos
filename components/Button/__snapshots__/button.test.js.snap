// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnchorButton/> renders correctly 1`] = `
<ButtonElement
  isHovered={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  An Anchor Button
</ButtonElement>
`;

exports[`<Button/> renders correctly 1`] = `
<ButtonElement
  isHovered={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  A Button
</ButtonElement>
`;
